app/models/goal.py

```diff
@@ -1,5 +1,36 @@
 from app import db
+from flask import Blueprint, make_response, request, jsonify, abort, request
+
 
 
 class Goal(db.Model):
```

Comment:
Though, I will mention that your base implementation for both your model classes, `Task` and `Goal` is very good overall. I like your pattern for writing the `update` methods, it's a great use of `except`.

Reply:
Aw, I just noticed you didn't get a chance to use the `update` methods in the routes. Well, just consider this a comment on how you're on the right track!